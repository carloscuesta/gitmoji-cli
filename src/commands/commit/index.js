// @flow
import inquirer from 'inquirer'

import getEmojis from '../../utils/getEmojis'
import prompts from './prompts'
import withHook from './withHook'
import withClient from './withClient'

const commit = (mode: 'client' | 'hook') => {
  return getEmojis()
    .then((gitmojis) => prompts(gitmojis))
    .then((questions) => {
      inquirer.prompt(questions).then((answers) => {
        if (mode === 'hook') return withHook(answers)

        return withClient(answers)
      })
    })
}

export default commit
