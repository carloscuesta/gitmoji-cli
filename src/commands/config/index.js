// @flow
import inquirer from 'inquirer'

import configurationPrompts, { CONFIGURATION_PROMPT_NAMES } from './prompts'
import configurationVault from '../../utils/configurationVault'

const config = () => {
  inquirer.prompt(configurationPrompts()).then((answers) => {
    configurationVault.setAutoAdd(answers[CONFIGURATION_PROMPT_NAMES.AUTO_ADD])
    configurationVault.setEmojiFormat(
      answers[CONFIGURATION_PROMPT_NAMES.EMOJI_FORMAT]
    )
    configurationVault.setSignedCommit(
      answers[CONFIGURATION_PROMPT_NAMES.SIGNED_COMMIT]
    )
    configurationVault.setScopePrompt(
      answers[CONFIGURATION_PROMPT_NAMES.SCOPE_PROMPT]
    )
  })
}

export default config
