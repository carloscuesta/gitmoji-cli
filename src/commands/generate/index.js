// @flow
import inquirer from 'inquirer'
import getEmojis from '@utils/getEmojis'
import getGeneratedMessage from '@utils/getGeneratedMessage'
import { execa } from 'execa'
import { type ParsedMessage } from '@utils/getGeneratedMessage'
import chalk from 'chalk'
import withClient from '@commands/commit/withClient'

type CommitOptions = {
  model?: 'gpt-3.5-turbo' | 'gpt-4',
  context?: boolean
}

const generate = async (options: CommitOptions): Promise<void> => {
  if (!process.env.OPENAI_API_KEY) {
    return console.error(
      chalk.red(`Unable to locate OPENAI_API_KEY in environment`)
    )
  }

  const { stdout: diff } = await execa('git', ['diff', '--cached'])
  if (!diff) {
    return console.error(chalk.red('Could not find any staged files to commit'))
  }

  // Let the user specify additional context to give some guidance to the generated message
  let context
  if (!!options.context) {
    ;({ context } = await inquirer.prompt(
      {
        type: 'text',
        name: 'context',
        message: 'Provide additional context for the commit message'
      },
      {
        onCancel: () => process.exit()
      }
    ))
  }

  let message: ParsedMessage | void
  const gitmojis = await getEmojis()
  while (true) {
    message = await getGeneratedMessage(diff, gitmojis, context, options.model)
    if (!message) {
      return
    }

    const { confirm } = await inquirer.prompt({
      type: 'confirm',
      name: 'confirm',
      message: message.raw
    })

    if (confirm) {
      break
    }
  }

  // Sanity should not be able to get here but Flow needed it
  if (!message) {
    return console.error(chalk.red('Unable to generate commit message'))
  }

  await withClient({
    title: message.title,
    gitmoji: message.gitmoji,
    message: ''
  })
}

export default generate
