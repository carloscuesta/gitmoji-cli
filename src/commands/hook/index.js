import create from './create'
import remove from './remove'

export default {
  create,
  remove
}
