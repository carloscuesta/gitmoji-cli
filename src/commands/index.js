const search = require('./search').default
const list = require('./list').default

export default {
  search,
  list
}
