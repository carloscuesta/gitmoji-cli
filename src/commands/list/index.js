// @flow
import getEmojis from '../../utils/getEmojis'
import printEmojis from '../../utils/printEmojis'

const list = () => {
  return getEmojis()
    .then((gitmojis) => printEmojis(gitmojis))
    .catch((err) => console.error(err))
}

export default list
