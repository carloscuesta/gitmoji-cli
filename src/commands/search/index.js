// @flow
import getEmojis from '../../utils/getEmojis'
import printEmojis from '../../utils/printEmojis'

const search = (query: string) => {
  return getEmojis()
    .then((gitmojis) =>
      gitmojis.filter((gitmoji) => {
        const emoji = gitmoji.name.concat(gitmoji.description).toLowerCase()
        return emoji.indexOf(query.toLowerCase()) !== -1
      })
    )
    .then((gitmojisFiltered) => printEmojis(gitmojisFiltered))
    .catch((err) => console.error(err))
}

export default search
