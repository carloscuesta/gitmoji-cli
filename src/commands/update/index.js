// @flow
import getEmojis from '../../utils/getEmojis'
import printEmojis from '../../utils/printEmojis'

const update = () => {
  return getEmojis(true)
    .then((gitmojis) => printEmojis(gitmojis))
    .catch((err) => console.error(err))
}

export default update
