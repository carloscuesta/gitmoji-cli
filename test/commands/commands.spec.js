import commands from '../../src/commands'

describe('commands', () => {
  it('should match commands', () => {
    expect(commands).toMatchSnapshot()
  })
})
