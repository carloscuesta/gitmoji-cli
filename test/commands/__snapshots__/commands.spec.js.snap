// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands should match commands 1`] = `
Object {
  "commit": [Function],
  "config": [Function],
  "createHook": [Function],
  "list": [Function],
  "removeHook": [Function],
  "search": [Function],
  "update": [Function],
}
`;
